datastore/statedata.go

```diff
@@ -237,9 +237,15 @@ type Artifact struct {
 	// of the Artifact structure for backwards compatability.
 	ArtifactName  string `json:"artifact_name"`
 	ArtifactGroup string `json:"artifact_group"`
-	// This property is only used for deferring parsing of artifacts of
-	// version v3 and later. Is only read as nil value.
+	// The following two properties together with CompatibleDevices
+	// implements the ArtifactDepends structure of artifact version 3.
+	// NOTE: these properties are not stored as part of StateData
+	ArtifactNameDepends  []string `json:"-"`
+	ArtifactGroupDepends []string `json:"-"`
```

Comment:
I don't think these belong here if we are not going to store them. The thing is that this data structure is reloaded from disk if the device crashes during the update. Therefore everything within it should always be storable.

We have access to the installer object in `UpdateStoreState.Handle()`, so later we can make the depends calculations there without having to store the depends in this struct.